llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2887,8 +2887,11 @@ SDValue DAGCombiner::visitADD(SDNode *N) {
 
   // fold (a+b) -> (a|b) iff a and b share no bits.
   if ((!LegalOperations || TLI.isOperationLegal(ISD::OR, VT)) &&
-      DAG.haveNoCommonBitsSet(N0, N1))
-    return DAG.getNode(ISD::OR, DL, VT, N0, N1);
+      DAG.haveNoCommonBitsSet(N0, N1)) {
+    SDNodeFlags Flags;
+    Flags.setDisjoint(true);
```

Comment:
Side note, we should fix the bad API of SDNodeFlags

Reply:
Yeah, this was gnarly but I just tried to follow what we do for nonneg